src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/AssemblyBuilderImpl.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace System.Reflection.Emit
+{
+    public sealed class AssemblyBuilderImpl : AssemblyBuilder
```

Comment:
Is this intentionally public?

Reply:
It is public in impl only to make the tests compile.

An alternative is to make this internal and switch tests to use reflection until we have approved public API and completed the implementation. Reflection would be only needed for AssemblyBuilderImpl constructor and Save method, so it is perfectly doable. Do you prefer that we go with internal + reflection for tests?

Reply:
> It is public in impl only to make the tests compile.
>
> An alternative is to make this internal and switch tests to use reflection until we have approved public API and completed the implementation.

Yes, it is public for testing, but not added to the ref, because not added to ref I am using reflection for testing, though I make this type internal causing type load exception, so kept it public https://github.com/dotnet/runtime/blob/3570054eca81355d0629fbafacead1ddca5410f0/src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblyTools.cs#L68-L70

Reply:
> though I make this type internal causing type load exception, 

You would have to add ILLink.Descriptors.LibraryBuild.xml to prevent the linker from stripping the implementation. For example, we do that here: https://github.com/dotnet/runtime/blob/8d2ddc45e8f6f2a726c2b91c1ba2b9f7208b1f3b/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.LibraryBuild.xml#L5

Reply:
I just assumed based on the name that we never intend for it to be public; it seems like it's intended to be an implementation detail.  If it's just for the ctor/Save and thus not a big hardship to go with the reflection-based approach, that would be my preference.